docs/types.rst

```diff
@@ -840,6 +849,81 @@ Of course, you can also directly access the members of the struct without
 assigning it to a local variable, as in
 ``campaigns[campaignID].amount = 0``.
 
+Types as input and output parameters
+====================================
+
+.. index:: !function parameters, functions, parameters
+
+Allowed function parameter types
+--------------------------------
+
+.. which types are allowed as function parameters depending on the kind of the function
+
+.. The types allowed as input or output function parameters depends on the kind of function you declare.
+
++------------------------------------------------+---------------------------------------------------------+-------------------------+
+| Type                                           | Input                                                   | Output                  |
```

Comment:
I'm not sure whether there is a difference between input and output.

Reply:
Perhaps better to drop the "allowed function parameter types" section from this PR.

Reply:
@chriseth Why do you say that? Wasn't that (half) the point of the PR? Or do you mean move it to a different PR?

And if there's no difference between input/output, then that makes it easier, so that's good :)

Reply:
If there is no logical rule behind something, a table is useful. If there is a logical rule behind something, then the rule should be stated and the table should only be used as an extra - you should not be required to derive the rule yourself.

Input and output is only relevant for reference types in external functions, so I don't think it makes sense to complicated the table just for that.